[Bug]: Android x86 builds return SSL Errors

### Contact Details

_No response_

### Version

5.4.0

### Description

Target Environment:

Android NDK 23c with `-march=i686 -msse3 -m32 -O3`.

The resulting build can no longer establish a TLS / DTLS connection. Using the option `--enable-heapmath` fixes the issue.

This was tested on the android emulator. 

Using a more modern x86_64 image does not exhibit the same issue when built with `-march=x86-64 -msse4.2 -mpopcnt -m64 -O3`

### Reproduction steps

_No response_

### Relevant log output

_No response_

Setting `SP_WORD_SIZE=32` does not seem to change anything.

Enabling fastmath results in the following error
```
wolfcrypt/src/tfm.c:3257:11: error: inline assembly requires more registers than available
          SQRADD(A->dp[ix>>1], A->dp[ix>>1]);
          ^
./wolfcrypt/src/asm.c:701:6: note: expanded from macro 'SQRADD'
     "movl  %3,%%eax     \n\t"                            \
     ^
wolfcrypt/src/tfm.c:3251:11: error: inline assembly requires more registers than available
          SQRADD2(*tmpx++, *tmpy--);
          ^
./wolfcrypt/src/asm.c:712:6: note: expanded from macro 'SQRADD2'
     "movl  %3,%%eax     \n\t"                            \
     ^
wolfcrypt/src/tfm.c:3251:11: error: inline assembly requires more registers than available
./wolfcrypt/src/asm.c:712:6: note: expanded from macro 'SQRADD2'
     "movl  %3,%%eax     \n\t"                            \
     ^
wolfcrypt/src/tfm.c:3251:11: error: inline assembly requires more registers than available
./wolfcrypt/src/asm.c:712:6: note: expanded from macro 'SQRADD2'
     "movl  %3,%%eax     \n\t"                            \
     ^
4 errors generated.
```

--- a/wolfcrypt/src/asm.c
+++ b/wolfcrypt/src/asm.c
@@ -703,7 +703,7 @@ __asm__(                                                  \
      "addl  %%eax,%0     \n\t"                            \
      "adcl  %%edx,%1     \n\t"                            \
      "adcl  $0,%2        \n\t"                            \
-     :"+rm"(c0), "+rm"(c1), "+rm"(c2)                     \
+     :"+m"(c0), "+m"(c1), "+m"(c2)                        \
      : "m"(i)                                             \
      :"%eax","%edx","cc");
 
@@ -717,7 +717,7 @@ __asm__(                                                  \
      "addl  %%eax,%0     \n\t"                            \
      "adcl  %%edx,%1     \n\t"                            \
      "adcl  $0,%2        \n\t"                            \
-     :"+rm"(c0), "+rm"(c1), "+rm"(c2)                     \
+     :"+m"(c0), "+m"(c1), "+m"(c2)                        \
      : "m"(i), "m"(j)                                     \
      :"%eax","%edx", "cc");
 

